src/lib/app.ts

```diff
@@ -183,12 +184,17 @@ export default async function getApp(
         res.send(indexHTML);
     });
 
-    app.get(`${baseUriPath}/*`, (req, res) => {
-        if (req.path.startsWith(`${baseUriPath}/api`)) {
-            res.status(404).send({ message: 'Not found' });
-            return;
-        }
+    // handle all API 404s
+    app.use(`${baseUriPath}/api`, (req, res) => {
+        const error = new UnleashError({
+            name: 'NotFoundError',
+            message: `The path you were looking for (${baseUriPath}/api${req.path}) is not available.`,
+        });
+        res.status(error.statusCode).send(error);
+        return;
+    });
```

Comment:
The idea here is that this should cover all API paths regardless of the method used (as opposed to the previous one that only covered GETs). I had some problems with this not working correctly, but I can't reproduce this now. If you don't think this is a good idea for some reason or if there is a better way to solve this, let me know.

Reply:
It's common in the middleware stack architecture for the 404 handler to be put just before the last error handler and pass the error with next(new NotFoundError()) and have all the error formatting in one error handler

Reply:
> Why not new NotFoundError? Why do you prefer to put the absence semantics in the name string vs the type itself?

Good question! This is actually sort of at the heart of this issue: I don't know whether we should ask people to always use the new error type or if they should keep using the old ones. We can use the new one to enforce that you pass in certain required data (such as errors, path and type, etc), but we might not want people to switch regardless.

> It's common in the middleware stack architecture for the 404 handler to be put just before the last error handler and pass the error with next(new NotFoundError()) and have all the error formatting in one error handler

Right! I'm not very familiar with express, so that's very good to know. I'm not sure where it'd go, then, but I'd be happy to take a look.

Reply:
Attempted to address this in d481fa02d4c7e0b5182e17257becd1c4bc63e37d. I would very much appreciate it if you had a look to make sure that it's done correctly 🙏🏼